poc/flp_generic.py

```diff
@@ -640,6 +641,140 @@ def test_vec_set_type_param(self, test_vec):
         return ['length', 'chunk_length']
 
 
+class MultiHotHistogram(Valid):
+    """
+    A validity circuit that checks each Client's measurement is a bit vector,
+    and has at most `max_num_bits` 1s.
+
+    In order to check the Client measurement `x` has at most `max_num_bits` 1s,
+    we ask the Client to send an additional `num_bits_for_count` bits that
+    encode the number of 1s in the measurement, with an offset. Specifically:
+    - Let `num_bits_for_count = int(log2(max_num_bits)) + 1`.
+    - Let `offset = 2**num_bits_for_count - 1 - max_num_bits`.
+    - Client will encode `count = offset + \sum_i x_i` in `num_bits_for_count`
+      bits.
+    - We can naturally bound `count` as the following:
+      `0 <= count <= 2**num_bits_for_count - 1`, and therefore:
+      `-offset <= \sum_i x_i <= max_num_bits`.
+    - Since we also verify each `x_i` is a bit, we can lower bound the summation
+      by 0. Therefore, we will be able to verify
+      `0 <= \sum_i x_i <= max_num_bits`.
+    """
+    # Operational parameters
+    length = None  # Set by `MultiHotHistogram.with_params()`
+    max_num_bits = None  # Set by `MultiHotHistogram.with_params()`
+    chunk_length = None  # Set by `MultiHotHistogram.with_params()`
+
+    # Associated types
+    Measurement = Unsigned
```

Comment:
What is this type? For `Histogram` it's a single bucket index, but for `MultiHotHistogram` we need to be able to represent multiple locations being set.

Reply:
Perhaps it should be a set of bucket indices, i.e., `set[Unsigned]`? That would line up better with the measurement type for `Histogram`.